pkg/controller/statefulset/stateful_pod_control.go

```diff
@@ -106,6 +106,13 @@ func (spc *realStatefulPodControl) UpdateStatefulPod(set *apps.StatefulSet, pod
 				return err
 			}
 		}
+
+		// if the Pod does not conform to the StatefulSet's nodeSelector, update the Pod's nodeSelector
+		if !nodeSelectorMatches(set, pod) {
+			updateNodeSelector(set, pod)
```

Comment:
In StatefulSet, it seems only update Pod's nodeSelector, who will trigger re-schedule?

Reply:
please correct me if I'm wrong but in `pkg/controller/statefulset/stateful_pod_control.go:122` following line will commit the update:

`_, updateErr := spc.client.CoreV1().Pods(set.Namespace).Update(pod)`

does it answer for your question?

Reply:
nop; this code only update pod object in apiserver; no one will trigger 'rescheduler'.